### PR TITLE
Improve Perceiver docs

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

Some last-minute changes to beautify the Perceiver docs.

### CODE REVIEW

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 1]
```diff
@@ -1188,10 +1189,11 @@ def forward(
 This model uses learned position embeddings. In other words, this model is not given any privileged information about
 the structure of images. As shown in the paper, this model can achieve a top-1 accuracy of 72.7 on ImageNet.
 
-`PerceiverForImageClassificationLearned` uses
-`transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "conv1x1") to
-preprocess the input images, and `transformers.models.perceiver.modeling_perceiver.PerceiverClassificationDecoder` to
-decode the latent representation of `~transformers.PerceiverModel` into classification logits.
+:class:`~transformers.PerceiverForImageClassificationLearned` uses
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "conv1x1") to
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1193]:
```suggestion
:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with :obj:`prep_type="conv1x1"`) to
```

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 2]
```diff
@@ -1326,10 +1328,11 @@ def forward(
 This model uses fixed 2D Fourier position embeddings. As shown in the paper, this model can achieve a top-1 accuracy of
 79.0 on ImageNet, and 84.5 when pre-trained on a large-scale dataset (i.e. JFT).
 
-`PerceiverForImageClassificationLearned` uses
-`transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "pixels") to
-preprocess the input images, and `transformers.models.perceiver.modeling_perceiver.PerceiverClassificationDecoder` to
-decode the latent representation of `~transformers.PerceiverModel` into classification logits.
+:class:`~transformers.PerceiverForImageClassificationLearned` uses
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "pixels") to
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1332]:
```suggestion
:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with :obj:`prep_type="pixels"`) to
```

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 3]
```diff
@@ -1461,10 +1464,11 @@ def forward(
 This model uses a 2D conv+maxpool preprocessing network. As shown in the paper, this model can achieve a top-1 accuracy
 of 82.1 on ImageNet.
 
-`PerceiverForImageClassificationLearned` uses
-`transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "conv") to preprocess
-the input images, and `transformers.models.perceiver.modeling_perceiver.PerceiverClassificationDecoder` to decode the
-latent representation of `~transformers.PerceiverModel` into classification logits.
+:class:`~transformers.PerceiverForImageClassificationLearned` uses
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "conv") to
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1468]:
```suggestion
:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with :obj:`prep_type="conv"`) to
```

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 4]
```diff
@@ -1592,10 +1596,11 @@ def forward(
 
 @add_start_docstrings(
     """
-Example use of Perceiver for optical flow, for tasks such as Sintel and KITTI. `PerceiverForOpticalFlow` uses
-`transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "patches") to
-preprocess the input images, and `transformers.models.perceiver.modeling_perceiver.PerceiverOpticalFlowDecoder` to
-decode the latent representation of `~transformers.PerceiverModel`.
+Example use of Perceiver for optical flow, for tasks such as Sintel and KITTI.
+:class:`~transformers.PerceiverForOpticalFlow` uses
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type` = "patches") to
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1601]:
```suggestion
:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverImagePreprocessor` (with `prep_type="patches"`) to
```

**Review by LysandreJik — APPROVED** [review]
Cool, LGTM!

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 5]
```diff
@@ -1717,25 +1722,26 @@ def forward(
...
-tensor into an actual video. It first splits up the output into the different modalities, and then applies the
+:class:`~transformers.PerceiverForMultimodalAutoencoding` uses
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverMultimodalPreprocessor` to preprocess the 3
+modalities: images, audio and class labels. This preprocessor uses modality-specific preprocessors to preprocess every
+modality separately, after which they are concatenated. Trainable position embeddings are used to pad each modality to
+the same number of channels to make concatenation along the time dimension possible. Next, one applies the Perceiver
+encoder.
+
+:class:`~transformers.models.perceiver.modeling_perceiver.PerceiverMultimodalDecoder` is used to decode the latent
+representation of :class:`PerceiverModel`. This decoder uses each modality-specific decoder to construct queries. The
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1733]:
```suggestion
representation of :class:`~transformers.PerceiverModel`. This decoder uses each modality-specific decoder to construct queries. The
```

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**